test/load_path_cache/cache_test.rb

```diff
@@ -165,9 +170,15 @@ def test_path_encoding
           require path
           internal_path = $LOADED_FEATURES.last
           assert_equal(OS_ASCII_PATH_ENCODING, internal_path.encoding)
-          assert_equal(OS_ASCII_PATH_ENCODING, path.encoding)
+          # TruffleRuby object is a copy and the encoding resets to utf-8.
+          assert_equal(OS_ASCII_PATH_ENCODING, path.encoding) unless truffleruby?
           File.write(path, "")
-          assert_same path, internal_path
+
+          if truffleruby?
+            assert_equal path, internal_path
```

Comment:
Is this comparison different from CRuby due to https://github.com/oracle/truffleruby/issues/3138?